test-dts/setupHelpers.test-d.ts

```diff
@@ -13,16 +13,20 @@ describe('defineProps w/ type declaration', () => {
   // type declaration
   const props = defineProps<{
     foo: string
+    bool?: boolean
   }>()
   // explicitly declared type should be refined
   expectType<string>(props.foo)
+  // the Boolean absent props will be cast to false
+  expectType<boolean>(props.bool)
```

Comment:
This implementation will lead it cannot jump to the declaration by clicking `bool` of `props.bool` (`props.foo` also).

Reply:
I'll create a new PR with another implementation.